jax/interpreters/xla.py

```diff
@@ -275,7 +275,9 @@ def abstractify(x) -> core.AbstractValue:
   raise TypeError(f"Argument '{x}' of type '{type(x)}' is not a valid JAX type")
 
 def _make_abstract_python_scalar(typ, val):
-  return ShapedArray((), dtypes._scalar_type_to_dtype(typ, val), weak_type=True)
+  # Note: all python scalar types are weak except bool
```

Comment:
Maybe add: because bool doesn't come in different sizes, so we don't need to worry about unwanted implicit type promtion [is this explanation correct??]

Reply:
Done!